src/plugins/cosmos/CosmosAsset.tsx

```diff
@@ -0,0 +1,40 @@
+import { Flex } from '@chakra-ui/react'
+import { useParams } from 'react-router-dom'
+import { Page } from 'components/Layout/Page'
+import { LoadingAsset } from 'pages/Assets/LoadingAsset'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import {
+  selectAssetByCAIP19,
+  selectMarketDataById,
+  selectMarketDataLoadingById
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { CosmosAssetAccountDetails } from './CosmosAssetAccountDetails'
+
+export const CosmosAsset = (props: { chainId: string }) => {
+  const dispatch = useAppDispatch()
+
+  const { assetSubId } = useParams<{ assetSubId: string }>()
+  const assetId = `${props.chainId}/${assetSubId}`
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  if (!marketData) dispatch(marketApi.endpoints.findByCaip19.initiate(assetId))
+
+  const loading = useAppSelector(state => selectMarketDataLoadingById(state, assetId))
+
+  return !asset || loading ? (
```

Comment:
Non-blocking but that can be simplified with a ternay in l.30 rendering one component or the other